hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/LegacyReplicationManager.java

```diff
@@ -398,7 +400,22 @@ protected void processContainer(ContainerInfo container,
            */
           if (replicas.isEmpty() && (container.getNumberOfKeys() == 0)) {
             closeEmptyContainer(container);
+            return;
+          }
+
+          if (!foundHealthy && container.getReplicationType().equals(
+              HddsProtos.ReplicationType.RATIS)) {
```

Comment:
Or actually, EC is not handled by the Legacy RM and the Ratis check here is just defensive, right? In that case just adding a comment explaining this might be good since it threw me off initially.

Reply:
> EC is not handled by the Legacy RM
You're right. A comment about EC containers elsewhere in that method threw me off and I didn't realise we don't even let EC containers be processed in legacy. I've removed that check entirely to avoid confusion.